core/src/codec.rs

```diff
@@ -1,18 +1,83 @@
 //! `Ipld` codecs.
-use crate::error::BlockError;
-use crate::ipld::Ipld;
-use std::error::Error;
+use std::io::{Read, Write};
 
 /// Codec trait.
-pub trait Codec {
+pub trait Codec: Sized {
     /// Codec version.
-    const VERSION: cid::Version;
+    const VERSION: cid::Version = cid::Version::V1;
```

Comment:
I'd remove the version. This is part of a CID and not of a Codec.

Reply:
good point